inc/components/post-timestamp/component.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Post timestamp.
+ *
+ * Get the post's timestamp(s).
+ *
+ * @package WP_Irving
+ */
+
+namespace WP_Irving\Components;
+
+use WP_Irving\Component;
+
+if ( ! function_exists( '\WP_Irving\get_registry' ) ) {
+	return;
+}
+
+/**
+ * Register the component and callback.
+ */
+\WP_Irving\get_registry()->register_component_from_config(
+	__DIR__ . '/component',
+	[
+		'callback' => function( Component $component ): Component {
+
+			// Get the post ID from a context provider.
+			$post_id = $component->get_config( 'post_id' );
+
+			$post = get_post( $post_id );
+			if ( ! $post instanceof \WP_Post ) {
+				return $component;
+			}
+
+			return $component->set_config(
+				'content',
+				sprintf(
+					/**
+					 * Translators:
+					 * %1$s - Published timestamp.
+					 * %2$s - Modified timestamp.
+					 */
+					$component->get_config( 'content_format' ),
```

Comment:
Interesting, I like this way of doing it, def provides flexibility

Reply:
I stumbled on the pattern building out the search. Gives the end user all the tools they need to build complex output without touching anything besides JSON. Pretty cool.